ldrshared/clients/messenger/base.py

```diff
@@ -0,0 +1,81 @@
+from abc import ABCMeta, abstractmethod
+from typing import Union
+
+
+class BaseMessage(metaclass=ABCMeta):
```

Comment:
It would be good to use generic types here. Also dataclasses are good

```python
from typing import TypeVar, Generic
from dataclasses import dataclass


T = TypeVar("T", bound=object) 

@dataclass
class BaseMessage(ABC, Generic[T]):
     message: T
     ....

@dataclass
class PubSubMessage(BaseMessage[Union[str, dict]]):
     ...
```

Then we always the implementing class defines the type of the `message` property in a *generic* way. We don't lose any of the lovely type checking goodness.

The type of `message` inside `PubSubMessage` would be `Union[str, dict]`.

Also the use of dataclasses here means we can avoid writing the `__init__` function entirely.